src/byond.rs

```diff
@@ -1,20 +1,21 @@
-use std::borrow::Cow;
-use std::cell::RefCell;
-use std::ffi::{CStr, CString};
-use std::slice;
-
-use std::os::raw::{c_char, c_int};
+use std::{
+    borrow::Cow,
+    cell::RefCell,
+    ffi::{CStr, CString},
+    os::raw::{c_char, c_int},
+    slice,
+};
 
 static EMPTY_STRING: c_char = 0;
 thread_local! {
     static RETURN_STRING: RefCell<CString> = RefCell::new(CString::default());
 }
 
-pub fn parse_args<'a>(argc: c_int, argv: &'a *const *const c_char) -> Vec<Cow<'a, str>> {
+pub fn parse_args<'a>(argc: c_int, argv: *const *const c_char) -> Vec<Cow<'a, str>> {
     unsafe {
         slice::from_raw_parts(*argv, argc as usize)
             .iter()
-            .map(|ptr| CStr::from_ptr(*ptr))
+            .map(|ptr| CStr::from_ptr(ptr))
```

Comment:
My fault as well for not catching it the first time but I'd like to point out now that this change was wrong. See 6ba75cd

Reply:
Weird, wonder what the actual solution to fixing this lint is then. I did build this for myself to try out and I didn't notice any issues, so maybe I just didn't test enough.